Data/Student.cs

```diff
@@ -0,0 +1,18 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace StudentGrades.Data;
+
+public class Student
+{
+    public int StudentId { get; set; }
+
+    [Required(ErrorMessage = "Please enter name")]
+    public string Name { get; set; } = "";
+
+    [Required(ErrorMessage = "Please enter subject")]
+    public string Subject { get; set; } = "";
+
+    [Required(ErrorMessage = "Please enter grade")]
+    [RegularExpression("^([1-9]|10)$", ErrorMessage = "Please enter a valid number from 0 to 10")]
+    public byte Grade { get; set; }
```

Comment:
```suggestion
    public byte? Grade { get; set; }
```
Для типа значений атрибут Required не нужен, они автоматически проверяются на непустоту. Однако и выводимым сообщением в случае пустой строки управлять нельзя.
![image](https://user-images.githubusercontent.com/3691630/209591856-ca774b64-ad7b-4ec6-ab81-594f00e1ad4b.png)
Самое простое решение - сделать тип ссылочным насильно.

P.S. Картинка из решения твоего одногруппника, но суть та же.